pkg/manifest/os.go

```diff
@@ -742,6 +742,13 @@ func (p *OS) serialize() osbuild.Pipeline {
 		}))
 	}
 
+	if osbuild.ContainsFIPSKernelOption(p.KernelOptionsAppend) {
+		fipsStage := osbuild.NewFIPSStage(&osbuild.FIPSStageOptions{
+			BootCfg: false,
+		})
+		pipeline.AddStage(fipsStage)
+	}
```

Comment:
Adding the stage based on the existence of the kernel argument is a bit unusual for the way we do things in the pipeline generators.  I can kind of see the logic behind this: the stage will be added even if the user sets `fips=false` in the blueprint (or omits it) but instead adds `fips=1` to the kernel options.  But since we're exposing it as a high level option, I don't think we need this "automatic" way of detecting its inclusion and we should instead use the user-provided to option for all conditions.

Let's instead add a `Fips bool` to the `OS struct` (and `OSTreeDeployment`) and make all changes based on that.

Reply:
I have been thinking a lot about this and I changed my mind and I would prefer to use the `fips=1` kernel argument as the user API for enabling FIPS mode to avoid this kind of confusion. This way we will be coherent and will maintain the current user API for all the supported FIPS installations.

Reply:
I understand your reasoning and motivation for this decision, but I'm not in favor of this approach for the following reasons:

- Blueprints are higher-level API and the customizations tend to abstract lower level configuration done to the image.
- Specifying `fips=1` on the kernel cmdline is not really sufficient to build image which uses FIPS mode. Based on this and the other PRs, one needs to potentially append the `boot=` option to kernel cmdline, set the crypto policy and modify dracut config. Gating "FIPS enabled image" customization behind just adding `fips=1` to the kernel cmdline IMO creates false expectations that this is really enough to have FIPS enables system.
- Users with older osbuild-composer, which won't support this feature, may still think that appending `fips=1` to kernel cmdline will do the trick. If there was an explicit option, they would get an error instead.

Reply:
> Specifying fips=1 on the kernel cmdline is not really sufficient to build image which uses FIPS mode. Based on this and the other PRs, one needs to potentially append the boot= option to kernel cmdline, set the crypto policy and modify dracut config

The idea here is that those things should *automatically follow* the `fips=1` karg; that's what the code in both Anaconda and RHEL CoreOS are doing.  I think it makes sense to do here too.